File: Back-End/src/controleDespesa.Application/DTOs/DespesaDTO.cs
using controleDespesa.Domain.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace controleDespesa.Application.DTOs
{
    public class DespesaDTO
    {

        public int Id { get; set; }

        public string Descricao { get; set; } = string.Empty;

        public decimal ValorDespesa { get; set; }

        public DateTime DataCadastro { get; set; }

        public DateTime? DataDespesa { get; set; }

      

        public bool DespesaFixa { get; set; } = false;

        public int TipoDespesaReceitaId { get; set; }
        [JsonIgnore]
        public TipoDespesaReceitaDTO? TipoDespesaReceita { get; set; }





    }
}

File: Front-End/controle-despesa/src/app/models/dtos/despesa.dto.ts
import { TipoDespesaReceitaResponse } from "../responses/tipo-despesa-receita-response";
import { TipoDespesaReceitaDTO } from "./tipoDespesaReceita.dto";

export interface DespesaDTO {
    id: number;
    descricao: string;
    tipoDespesaReceitaId: number | null;
    numeroDeParcela: number;
    valorDespesa: number | null;
    dataCadastro: Date;
    dataDespesa: Date;
    parcelado: boolean;
    despesaFixa: boolean;
    usuarioId:number;
    tipoDespesaReceitaDTO?: TipoDespesaReceitaDTO;



}

